tensorflow/core/framework/tensor.cc

```diff
@@ -1016,13 +1016,13 @@ Tensor Tensor::SubSlice(int64_t index) const {
   CHECK_GE(dims(), 1);  // Crash ok.
   CHECK_LE(0, index);   // Crash ok.
   int64_t dim0_size = shape_.dim_size(0);
-  CHECK_LE(index, dim0_size);  // Crash ok.
   Tensor ret;
   ret.shape_ = shape_;
   ret.shape_.RemoveDim(0);
   ret.set_dtype(dtype());
   ret.buf_ = nullptr;
   if (dim0_size > 0) {
+    CHECK_LT(index, dim0_size);  // Crash ok.
```

Comment:
The case of `dim0_size == 0` but `index > 0` should fail.  It looks like it returns an invalid tensor. 
 Consider a tensor with shape `[0, 5, 5, 5]`.  This is empty and the buffer pointer would be `nullptr`.  This code as written will return a tensor of size `[5, 5, 5]` but a `nullptr` buffer.

In fact, whenever `index >= dim0_size`, we would get an invalid tensor, even if `dim0_size > 0`.  It's an invalid index, so we can't chip away a subtensor.

The original check should be `CHECK_LT(index, dim0_size)`.

Reply:
I guess technically `Slice` is also incorrect for the case where `start == limit == dim0_size`.  Not only do we need `0 <= start <= limit <= dim0_size`, we also need `start < dim0_size`.  If you give it a tensor of `[2, 2]` and try to `Slice(2, 2)`, it currently looks like it returns an empty tensor, but really you've gone past the end of the tensor, so it should probably fail instead.